viite-backend/base/src/main/scala/fi/vaylavirasto/viite/model/RoadPart.scala

```diff
@@ -0,0 +1,48 @@
+package fi.vaylavirasto.viite.model
+
+import scala.math.Ordered.orderingToOrdered // for compare function
+
+/**
+ * RoadPart contains road number, and part number information of a road, providing basic validation
+ * for the numbers' correctness, and basic boolean functions to easily check the state of the RoadPart.
+ *
+ * If a RoadPart has either or both its values 0, that is, e.g. RoadPart(0,0), the RoadPart is said to be unaddressed, and/or invalid.
+ *
+ * @throws IllegalArgumentException at construction, if the roadNumber, or partNumber are out of acceptable values.
+ */
+case class RoadPart (roadNumber: Long, partNumber: Long) extends Ordered[RoadPart]
+{
+
+  val maxRoadNumber = 99999
+  val maxPartNumber = 999
+
+  // throws IllegalArgumentException if requirements not met
+  require((roadNumber>=0 & roadNumber <= maxRoadNumber), s"Road number must be between (0-)1-$maxRoadNumber")
+  require((partNumber>=0 & partNumber <= maxPartNumber), s"Road number must be between (0-)1-$maxPartNumber")
+
+  /** Returns the RoadPart in the format "roadNumber/partNumber", e.g. "3575/1".
+    * Overrides the very basic java.Object.toString.
+    * @override [[java.object.String]] */
+  override def toString : String = {  s"$roadNumber/$partNumber"  }
+
+  /** Returns true, if the compared RoadParts have same road number, and both RoadParts are valid. Else false. */
+  def isAtSameRoadThan(other: RoadPart): Boolean = {  this.roadNumber == other.roadNumber && this.isValid && other.isValid }
```

Comment:
NIL: Tää "than" kuulostaa hassulta omaan korvaan, oisko "with" parempi sana vaihtoehto?

Reply:
🤔 Kieliopillisesti toi on hyvinkin oikein, mutta Sanakirja.org:in mukaan kuuluu luokkaan "Vanhentuneet sanat", siksi kenties kuulostaa hassulta 😄 
Mulle se kyllä (näköjään) on ainoa oikea, mut muutos lienee tarpeen.... 👌 

Bing co-pilot sanoi näin:
“Sama kuin X” englanniksi on “the same as X”

Reply:
isAtSameRoadAs(other) on hyvä

Reply:
FIksattu